mg/builtins.go

```diff
@@ -31,14 +32,19 @@ func (bcl BultinCmdList) Lookup(name string) (cmd BultinCmd, found bool) {
 	panic("internal error: the `.exec` BuiltinCmd is not defined")
 }
 
+// BuiltinCmds is used to add a list of predefined commands to the state. It
+// implements Reducer interface.
 type BuiltinCmds struct{}
 
+// ExecCmd spawns the given command in a goroutine and returns its State right
+// away. If the name is `.exec`, it will use its first argument as the `Name`
```

Comment:
No need to add comments like these, they're just stating what the implementation does and you'd then need to change the docs every time the implementation changes.